### PR TITLE
[Feature] Support classwise weight in losses

### PR DESCRIPTION
**Pull request by Ezra-Yu**
1. add classwise_weight in losses like CE,BCE,SoftCE.
2. add some unit test

refer to #379

### CODE REVIEW

**Comment by mzr1996** [comment]
Wait for https://github.com/open-mmlab/mmdetection/pull/5776

File: tests/test_metrics/test_losses.py [hunk 1]
```diff
@@ -45,38 +45,69 @@ def test_cross_entropy_loss():
...
+    assert torch.allclose(loss(cls_score, label), torch.tensor(176.667))
+    # test bce_loss with weight
+    assert torch.allclose(
+        loss(cls_score, label, weight=weight), torch.tensor(74.333))
 
     # test soft_ce_loss
     cls_score = torch.Tensor([[100, -100]])
     label = torch.Tensor([[1, 0], [0, 1]])
     weight = torch.tensor(0.5)
+    class_weight = torch.tensor(0.25)
```
Comment (mzr1996) [inline tests/test_metrics/test_losses.py:108]: In `soft_cross_entropy`, `cls_score` should share the same shape with `label`, why they're different here.
And according to `label`, there should be two samples, but `weight` only has one item.

**Review by mzr1996 — CHANGES_REQUESTED** [review]

File: mmcls/models/losses/cross_entropy_loss.py [hunk 2]
```diff
@@ -148,10 +172,18 @@ def forward(self,
         assert reduction_override in (None, 'none', 'mean', 'sum')
         reduction = (
             reduction_override if reduction_override else self.reduction)
+
+        if self.class_weight is not None:
+            class_weight = cls_score.new_tensor(
+                self.class_weight, device=cls_score.device)
```
[the remark this answers is not in the document]
Comment (mzr1996) [reply mmcls/models/losses/cross_entropy_loss.py:178]: Here I think we can to `to` function refers to [the docs](https://pytorch.org/docs/stable/generated/torch.Tensor.to.html)

**Review by mzr1996 — APPROVED** [review]
LGTM
